src/main/java/teammates/sqllogic/core/UsageStatisticsLogic.java

```diff
@@ -0,0 +1,64 @@
+package teammates.sqllogic.core;
+
+import java.time.Instant;
+import java.util.List;
+
+import teammates.storage.sqlapi.UsageStatisticsDb;
+import teammates.storage.sqlentity.UsageStatistics;
+
+/**
+ * Handles operations related to system usage statistics objects.
+ *
+ * @see UsageStatistics
+ * @see teammates.storage.api.UsageStatisticsDb
+ */
+public final class UsageStatisticsLogic {
+
+    private static final UsageStatisticsLogic instance = new UsageStatisticsLogic();
+
+    private final UsageStatisticsDb usageStatisticsDb = UsageStatisticsDb.inst();
+
+    private UsageStatisticsLogic() {
+        // prevent initialization
+    }
+
+    public static UsageStatisticsLogic inst() {
+        return instance;
+    }
+
+    void initLogicDependencies() {
+        // TODO
+    }
+
+    /**
+     * Gets the list of statistics objects between start time and end time.
+     */
+    public List<UsageStatistics> getUsageStatisticsForTimeRange(Instant startTime, Instant endTime) {
+        return usageStatisticsDb.getUsageStatisticsForTimeRange(startTime, endTime);
+    }
+
+    /**
+     * Calculates the usage statistics of created entities for the given time range.
+     */
+    public UsageStatistics calculateEntitiesStatisticsForTimeRange(Instant startTime, Instant endTime) {
+        int numResponses = 0; //feedbackResponsesLogic.getNumFeedbackResponsesByTimeRange(startTime, endTime);
+        int numCourses = 0; //coursesLogic.getNumCoursesByTimeRange(startTime, endTime);
+        int numStudents = 0; //studentsLogic.getNumStudentsByTimeRange(startTime, endTime);
```

Comment:
Note that we will need to support the old methods as well. Course related entities will not be migrated immediately, so there will be a period of time we have to fetch statistics for both new and old entities.

Reply:
@samuelfangjw I will make sure to do this in `CalculateUsageStatisticsAction` in the next PR